extensions/shared/external-media/external-media-modal.js

```diff
@@ -0,0 +1,245 @@
+/**
```

Comment:
FYI, this is a refactored version of `withMedia` HOC. 90% of the original code has been copied over. It's just a standalone component now and uses the editor state for the modal visibility toggle.